tests/test_spidermiddleware_offsite.py

```diff
@@ -55,21 +55,21 @@ def test_process_spider_output(self):
 
 class TestOffsiteMiddleware3(TestOffsiteMiddleware2):
 
-    def _get_spider(self):
-        return Spider('foo')
+    def _get_spiderargs(self):
+        return dict(name='foo')
 
```

Comment:
Just an aside. Does this class even do anything? Or could we clean this up?

Reply:
This is already the clean up! :)

The purpose of the class was to test that not setting the "allowed_domains" attribute in a spider works. However, due to a refactoring, the test was effectively a no-op because `_get_spider()` was never executed. My change should fix the test so it actually tests that use case again.

Reply:
Ah, of course. I blanked out the subclassing of `TestOffsiteMiddleware2` somehow. Looks good.